### PR TITLE
fix(hs): time stamp and contact not found issue

### PR DESCRIPTION
**Pull request by mihir-4116**
## Description of the change

> Description here

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

File: src/v0/destinations/hs/HSTransform-v1.js [hunk 1]
```diff
@@ -80,6 +82,21 @@ const processLegacyIdentify = async (message, destination, propertyMap) => {
       )}/${hsSearchId}`;
       response.method = defaultPatchRequestConfig.requestMethod;
     }
+    
+    if (!propertyMap) {
```
Comment (koladilip) [inline src/v0/destinations/hs/HSTransform-v1.js:86]: repeated code should move to common function

File: src/v0/destinations/hs/util.js [hunk 2]
```diff
@@ -478,8 +487,16 @@ const getExistingData = async (inputs, destination) => {
...
     const { message } = input;
     const { destinationExternalId } = getDestinationExternalIDInfoForRetl(message, DESTINATION);
-    values.push(destinationExternalId);
+
+    if(typeof destinationExternalId === 'string'){
+      values.push(destinationExternalId.toLowerCase());
+    } else{
+      const value = typeof destinationExternalId === 'object' ? JSON.stringify(destinationExternalId).toLowerCase() : destinationExternalId.toString();
+      values.push(value);
+    }
```
Comment (utsabc) [inline src/v0/destinations/hs/util.js:496]: No need of this we would always expect external Id to be something simple like number or string.
Just do
```
values.push(destinationExternalId.toString().toLowerCase());
```

File: src/v0/destinations/hs/HSTransform-v1.js (hunk 1 above)
Comment (mihir-4116) [reply src/v0/destinations/hs/HSTransform-v1.js]: updated

File: src/v0/destinations/hs/util.js (hunk 2 above)
Comment (mihir-4116) [reply src/v0/destinations/hs/util.js]: updated

**Review by koladilip — APPROVED** [review]

**Review by krishna2020 — APPROVED** [review]

File: src/v0/destinations/hs/util.js [hunk 3]
```diff
@@ -626,6 +635,31 @@ const getHsSearchId = (message) => {
   return { hsSearchId };
 };
 
+/**
+ * returns updated traits
+ * @param {*} propertyMap 
+ * @param {*} traits 
+ * @param {*} destination 
+ */
```
Comment (krishna2020) [inline src/v0/destinations/hs/util.js:643]: unit test is missing

Comment (mihir-4116) [reply src/v0/destinations/hs/util.js:643]: will add it as backlog, as of now we are going with hotfix

**Review by ruchiramoitra — APPROVED** [review]

**Pull request merged**